main.go

```diff
@@ -305,14 +307,19 @@ func getHealthCheckURL(endpoint, healthCheckPath string, healthCheckPort int) (s
 }
 
 // healthCheck - background routine which checks if a backend is up or down.
-func (b *Backend) healthCheck() {
+func (b *Backend) healthCheck(ctxt context.Context) {
+	ticker := time.NewTicker(b.healthCheckDuration)
+	defer ticker.Stop()
 	for {
-		err := b.doHealthCheck()
-		if err != nil {
-			console.Fatalln(err)
+		select {
+		case <-ctxt.Done():
+			return
+		case <-ticker.C:
+			err := b.doHealthCheck()
+			if err != nil {
+				console.Errorln(err)
+			}
```

Comment:
```suggestion
			err := b.doHealthCheck()
			if err != nil {
				console.Errorln(err)
			}
```

We should convert this into a counter to indicate "how many times healthcheck" failed instead of logging errors IMO.